cirq-google/cirq_google/json_resolver_cache.py

```diff
@@ -33,7 +33,7 @@ def _class_resolver_dictionary() -> Dict[str, ObjectFactory]:
         'GoogleNoiseProperties': cirq_google.GoogleNoiseProperties,
         'SycamoreGate': cirq_google.SycamoreGate,
         'GateTabulation': cirq_google.GateTabulation,
-        'GridDevice': cirq_google.GridDevice,
+        'GoogleGridDevice': cirq_google.GoogleGridDevice,
```

Comment:
This breaks old objects serialized with the `GridDevice` name. For backwards compatibility, the old mapping of 'GridDevice': GoogleGridDevice must also be preserved.

Reply:
Does this mean we can never add a new serializable `GridDevice` in the future ?

Reply:
> Does this mean we can never add a new serializable `GridDevice` in the future ?

That's how I understand it. For better or for worse, the change to using class names as the `cirq_type` locks us out of recycling names indefinitely.

Reply:
@verult Do you still think there's value in doing this rename?

Reply:
As discussed offline with @95-martin-orion , unfortunately we don't include the module name in our JSON serialization, so if there is ever a `cirq.GridDevice`, it would be indistinguishable from the cirq_google one for JSON deserialization. So yes, I'll move forward with this after addressing Orion's serialization backward compatibility concerns.

Reply:
Synced with Cheng - the rename is safe since this is all happening within a single release cycle (0.14-0.15).

Reply:
Synced with @95-martin-orion on the backward compatibility issue. Since GridDevice did not exist prior to the last release, it won't be necessary to keep an extra backward compatible mapping.